src/settings.ts

```diff
@@ -63,5 +67,85 @@ export class LinkRangeSettingTab extends PluginSettingTab {
 					await this.plugin.saveSettings();
 					postProcessorUpdate(this.app)
 				}));
+	
+		new Setting(this.containerEl)
+			.setName("Add New")
+			.setDesc("Add new pattern to match files in a directory. The first value will change the visual for the heading in a link. The second value will change the visual for separator. The third specifies the folder in which the files must be to match.")
+			.addButton((button: ButtonComponent) => {
+				button
+					.setTooltip("Add new pattern to match files in a directory.")
+					.setButtonText("+")
+					.setCta()
+					.onClick(() => {
+						this.plugin.settings.patterns.push({
+							headingVisual: '',
+							headingSeparatorVisual: '',
+							path: ''
+						});
+						this.plugin.saveSettings();
+						this.display();
+					});
+			});
+		
+			this.plugin.settings.patterns.forEach(
+				(pattern, index) => {
+					const s = new Setting(this.containerEl)
+					.addText(text => text
+						.setPlaceholder('Enter a heading override')
+						.setValue(pattern.headingVisual)
+						.onChange(async (value) => {
+							pattern.headingVisual = value;
+							await this.plugin.saveSettings();
+							postProcessorUpdate(this.app)
+						}))
+					.addText(text => text
+						.setPlaceholder('Enter a separator override')
+						.setValue(pattern.headingSeparatorVisual)
+						.onChange(async (value) => {
+							pattern.headingSeparatorVisual = value;
+							await this.plugin.saveSettings();
+							postProcessorUpdate(this.app)
+						}))
+					.addText(text => text
+						.setPlaceholder('Enter a path')
```

Comment:
If the first row is going to be the default pattern for the entire vault, I'd think we want to disable the path field just to make that clear, and/or set it to "/".

Reply:
Disabled with placeholder text explaining that it's the vault global